core/templates/dev/head/editor/EditorServices.js

```diff
@@ -1805,17 +1804,23 @@ oppia.factory('explorationWarningsService', [
 
   var _updateWarningsList = function() {
     _warningsList = [];
+    var _statesWithErrors = [];
 
     graphDataService.recompute();
     var _graphData = graphDataService.getGraphData();
 
     var statesWithoutInteractionIds = _getStatesWithoutInteractionIds();
     if (statesWithoutInteractionIds.length) {
-      _warningsList.push({
-        type: WARNING_TYPES.ERROR,
-        message: (
-          'Please add interactions for these cards: ' +
-          statesWithoutInteractionIds.join(', ') + '.')
+      angular.forEach(statesWithoutInteractionIds, function(
+        stateWithoutInteractionIds) {
+        if (_statesWithErrors.indexOf(stateWithoutInteractionIds) === -1) {
+          _statesWithErrors.push(stateWithoutInteractionIds);
+        }
+        _perStateErrors.push({
```

Comment:
I'm not sure _perStateErrors needs a warning type. Also I don't think the error messages need to include the name of the card.